drivers: peci: user space handlers not building correctly

**Describe the bug**
PECI subsystem cannot be enabled with USERSPACE feature 
User handlers do not build correctly.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - ...

**To Reproduce**
Steps to reproduce the behavior:
1) cd samples/drivers/peci
2)  west build -b mec172xevb_assy6906  --  -DCONFIG_ARM_MPU=y -DCONFIG_USERSPACE=y
3) See build error

**Expected behavior**
PECI handlers build correctly

**Impact**
High

**Logs and console output**
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c:14:9: error: implicit declaration of function 'Z_SYSCALL_DRIVER_PECI'; did you mean 'Z_SYSCALL_DRIVER_GEN'? [-Werror=implicit-function-declaration]
   14 |  Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, config));
      |         ^~~~~~~~~~~~~~~~~~~~~
/home/jamezaar/ecwork/zephyr_fork/include/syscall_handler.h:293:7: note: in definition of macro 'Z_OOPS'
  293 |   if (expr) { \
      |       ^~~~
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c:14:36: error: 'config' undeclared (first use in this function); did you mean 'mpu_config'?
   14 |  Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, config));
      |                                    ^~~~~~
/home/jamezaar/ecwork/zephyr_fork/include/syscall_handler.h:293:7: note: in definition of macro 'Z_OOPS'
  293 |   if (expr) { \
      |       ^~~~
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c:14:36: note: each undeclared identifier is reported only once for each function it appears in
   14 |  Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, config));
      |                                    ^~~~~~
/home/jamezaar/ecwork/zephyr_fork/include/syscall_handler.h:293:7: note: in definition of macro 'Z_OOPS'
  293 |   if (expr) { \
      |       ^~~~
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c: In function 'z_vrfy_peci_enable':
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c:22:36: error: 'enable' undeclared (first use in this function)
   22 |  Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, enable));
      |                                    ^~~~~~
/home/jamezaar/ecwork/zephyr_fork/include/syscall_handler.h:293:7: note: in definition of macro 'Z_OOPS'
  293 |   if (expr) { \
      |       ^~~~
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c: In function 'z_vrfy_peci_disable':
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c:30:36: error: 'disable' undeclared (first use in this function)
   30 |  Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, disable));
      |                                    ^~~~~~~
/home/jamezaar/ecwork/zephyr_fork/include/syscall_handler.h:293:7: note: in definition of macro 'Z_OOPS'
  293 |   if (expr) { \
      |       ^~~~
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c: In function 'z_vrfy_peci_transfer':
/home/jamezaar/ecwork/zephyr_fork/drivers/peci/peci_handlers.c:41:36: error: 'transfer' undeclared (first use in this function)
   41 |  Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, transfer));
      |                                    ^~~~~~~~
/home/jamezaar/ecwork/zephyr_fork/include/syscall_handler.h:293:7: note: in definition of macro 'Z_OOPS'
  293 |   if (expr) { \

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

--- a/include/zephyr/drivers/peci.h
+++ b/include/zephyr/drivers/peci.h
@@ -199,7 +199,7 @@ typedef int (*peci_transfer_t)(const struct device *dev, struct peci_msg *msg);
 typedef int (*peci_disable_t)(const struct device *dev);
 typedef int (*peci_enable_t)(const struct device *dev);
 
-struct peci_driver_api {
+__subsystem struct peci_driver_api {
 	peci_config_t config;
 	peci_disable_t disable;
 	peci_enable_t enable;

--- a/samples/drivers/peci/boards/mec172xevb_assy6906.overlay
+++ b/samples/drivers/peci/boards/mec172xevb_assy6906.overlay
@@ -1,0 +1,10 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/ {
+	aliases {
+		peci-0 = &peci0;
+	};
+};

